src/test/java/com/stripe/model/PagingIteratorTest.java

```diff
@@ -77,6 +95,56 @@ public PageableModelCollection answer(InvocationOnMock invocation) {
             Mockito.<RequestOptions>any());
   }
 
+  @Test
+  void testAutoPaginationFromReferencedCollection() throws StripeException, IOException {
+    Mockito.doAnswer(
+            new Answer<ReferencesPageableModel>() {
+              public ReferencesPageableModel answer(InvocationOnMock invocation) throws Exception {
+                return ApiResource.GSON.fromJson("{\"id\": \"xyz\", \"pages\": {\"data\": [{\"id\": \"pm_121\"}, {\"id\": \"pm_122\"}], \"url\": \"/v1/pageable_models\", \"has_more\": true}}", ReferencesPageableModel.class);
+              }
+            })
+        .when(networkSpy)
+        .request(
+            Mockito.any(ApiResource.RequestMethod.class),
+            Mockito.anyString(),
+            Mockito.<Map<String, Object>>any(),
+            Mockito.<Class<ReferencesPageableModel>>any(),
+            Mockito.<RequestOptions>any());
+    Stripe.apiKey = null;
+    ReferencesPageableModel model =
+        ReferencesPageableModel.retrieve(RequestOptions.builder().setApiKey("sk_test_xyz").build());
```

Comment:
Should we assert that RequestOptions flow/don't flow?

Reply:
I think a couple lines down we assert that they DON'T flow in this case, but I added a comment to make this more explicit, and I also added the assertion to the other test case where they DO flow.